harvard-huit/Harvard-Geospatial-Library
# Switching JS compression from Uglifier to Terser.

**Switching JS compression from Uglifier to Terser.**
* * *

* Other Relevant Links (Mailing list discussion, related pull requests, etc.)
- https://github.com/ahorek/terser-ruby
- https://github.com/lautis/uglifier/issues/142

# What does this Pull Request do?
When running HGL in "Test" mode it throws ES6 errors when Uglifier tries to build the JavaScript assets. This PR replaces Uglifier with Terser which is ES6 compatible.  The Uglifier solution we use now with the "Harmony" option is no longer supported according to their issue queue (see Relevant Link above) and Terser is the recommended replacement.

# How should this be tested?

A description of what steps someone could take to:
* Switch your branch to `main`
* Set your environment to "test" mode by changing your `.env` and `webapp.conf` files to use test (see next two steps):
* In `.env`: `RAILS_ENV=test`
* In `webapp.conf`: `passenger_app_env test;`
* Rebuild the container. When you load HGL in your browser, it will throw a generic internal server error.
* Switch and pull in the `fix-js-compressor` branch
* Completely rebuild the container
* HGL should function normally in your browser.

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? No 
- integration tests? No 

# Interested parties
@enriquediaz @dl-maura
